libelement.CLI/include/compile_command.hpp

```diff
@@ -0,0 +1,283 @@
+#pragma once
+
+#include <element/element.h>
+#include <interpreter_internal.hpp>
+#include <iostream>
+#include <fstream>
+
+#include "command.hpp"
+#include "compiler_message.hpp"
+
+#include "lmnt/opcodes.h"
+#include "lmnt/archive.h"
+#include "lmnt/interpreter.h"
+#include "lmnt/compiler.hpp"
+#include "lmnt/jit.h"
+
+namespace libelement::cli
+{
+struct compile_command_arguments
+{
+    std::string name;
+    std::string parameters;
+    std::string return_type;
+    std::string expression;
+    std::string output_path;
+
+    [[nodiscard]] std::string as_string() const
+    {
+        std::stringstream ss;
+        ss  << "compile --name \"" << name
+            << "\" --return-type \"" << return_type
+            << "\" --expression \"" << expression
+            << "\" --output-path \"" << output_path << "\"";
+        if (!parameters.empty())
+            ss << " --parameters \"" << parameters << "\"";
+        return ss.str();
+    }
+};
+
+class compile_command final : public command
+{
+public:
+    compile_command(common_command_arguments common_arguments,
+        compile_command_arguments custom_arguments)
+        : command(std::move(common_arguments))
+        , custom_arguments{ std::move(custom_arguments) }
+    {}
+
+    [[nodiscard]] compiler_message execute(const compilation_input& compilation_input) const override
+    {
+        const auto result = setup(compilation_input);
+        if (result != ELEMENT_OK)
+            return compiler_message(error_conversion(result),
+                "Failed to setup context",
+                compilation_input.get_log_json()); // todo
+
+        return compile(compilation_input);
+    }
+
+    [[nodiscard]] compiler_message compile(const compilation_input& compilation_input) const
+    {
+        const auto def = fmt::format("{}{}:{} = {}",
+            custom_arguments.name,
+            custom_arguments.parameters,
+            custom_arguments.return_type,
+            custom_arguments.expression);
+        element_result result = element_interpreter_load_string(context, def.c_str(), "<cli>");
+        if (result != ELEMENT_OK) {
+            return compiler_message(error_conversion(result),
+                "Failed to load: " + def + " at runtime with element_result " + std::to_string(result),
+                compilation_input.get_log_json());
+        }
+
+        constexpr auto max_output_size = 512;
```

Comment:
looks like this isn't used, both here and in the original command :b could be removed 🤷